start.js

```diff
@@ -0,0 +1,74 @@
+const http = require("http");
+const soldierRoute = require("./routes/soldiers.js");
+const dutiesRoute = require("./routes/duties");
+const justiceBoardRoute = require("./routes/justiceBoard");
+const configs = require('./configFile');
+const MongoClient = require('mongodb').MongoClient;
+
+// Connection URL
+const mongoUrl = configs.dbUrl;
+
+// Database Name
+const dbName = configs.dbName;
+
+function routesCases(res, req, primeFolder, numFolders, db, method, url, body, resultCallback) {
+  if (/soldiers.*/.test(primeFolder) && numFolders <= 3) {
+    soldierRoute.soldierRequest(db, method, url, body, resultCallback);
+  } else if (/duties.*/.test(primeFolder) && numFolders <= 4) {
+    if (method == "OPTIONS") {
+      method = req.headers["access-control-request-method"];
+    }
+    dutiesRoute.dutieRequest(db, method, url, body, resultCallback);
+  } else if (/justiceBoard/.test(url) && method == "GET") {
+    justiceBoardRoute.showBoard(db, resultCallback)
+  } else {
+    res.statusCode = 404;
+    res.end();
+  }
+}
+
+function onEndResponse(req, res, db, method, url, body) {
+  body = Buffer.concat(body).toString();
+  const numFolders = url.split("/").length;
+  const primeFolder = url.split("/")[1];
+  const resultCallback = function (err, result, statusCode) {
+    if (err) {
+      res.statusCode = statusCode;
+      res.end(err);
+    } else {
+      res.end(result);
+    }
+  };
+
+  if (body != "") {
+    body = JSON.parse(body);
+  }
+
+  routesCases(res, req, primeFolder, numFolders, db, method, url, body, resultCallback);
```

Comment:
if this routes the cases...why is it getting the req and res? it shouldn't be aware what those are. only recieve what it needs, you are leaving the server now moving to a different encapsulation.

Reply:
most of these parameters seem to be configurations and therefor shouldn't be parameters in calling the method everytime. initialize your router first with these parameters and clean this method signature. have you read clean code chapter 1-5? these are way too many parameters